### PR TITLE
support compile for random gaussian blur

### PR DESCRIPTION
**Pull request by edgarriba**
- support compile for random gaussian blur
- some fixes from https://github.com/kornia/kornia/pull/2865

#### Type of change

- [ ] 📚  Documentation Update
- [ ] 🧪 Tests Cases
- [ ] 🐞 Bug fix (non-breaking change which fixes an issue)
- [ ] 🔬 New feature (non-breaking change which adds functionality)
- [ ] 🚨 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] 📝 This change requires a documentation update

#### Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] Did you update CHANGELOG in case of a major change?

### CODE REVIEW

File: kornia/augmentation/_2d/intensity/gaussian_blur.py [hunk 1]
```diff
@@ -61,28 +61,51 @@ def __init__(
...
-
-        self.flags = {"kernel_size": kernel_size, "separable": separable, "border_type": BorderType.get(border_type)}
+        self.flags = {
+            "kernel_size": kernel_size,
+            "separable": separable,
+            "border_type": BorderType.get(border_type),
+        }
         self._param_generator = rg.RandomGaussianBlurGenerator(sigma)
 
+        self.gaussian_blur2d_fn = gaussian_blur2d
```
Comment (johnnv1) [inline kornia/augmentation/_2d/intensity/gaussian_blur.py:74]:
```suggestion
        self._gaussian_blur2d_fn = gaussian_blur2d
```

**Review by johnnv1 — APPROVED** [review]
LGTM, just move the gauss fn to be private

File: tests/augmentation/test_augmentation.py [hunk 2]
```diff
@@ -3881,6 +3847,19 @@ def test_module_sigma_tensor(self, device, dtype):
...
+        torch_version_le(1, 13, 1),
+        reason="torch.compile is not available in previous versions",
+    )
+    def test_compile(self, device, dtype):
+        kernel_size = (3, 3)
+        sigma = (1.5, 2.1)
+        img = torch.rand(1, 3, 5, 5, device=device, dtype=dtype)
+        aug = RandomGaussianBlur(kernel_size, sigma, "replicate")
+        aug = aug.compile(fullgraph=True)
+        assert aug(img).shape == img.shape
```
Comment (johnnv1) [inline tests/augmentation/test_augmentation.py:3861]:
```suggestion
        aug = RandomGaussianBlur(kernel_size, sigma, "replicate")
        expected = aug(img)
        aug = aug.compile(fullgraph=True)
        actual = aug(img)
        assert actual.shape == img.shape
        self.assert_close(expected, actual)
```

File: tests/augmentation/test_augmentation.py [hunk 3]
```diff
@@ -3881,6 +3847,22 @@ def test_module_sigma_tensor(self, device, dtype):
...
+    def test_compile(self, device, dtype):
+        kernel_size = (3, 3)
+        sigma = (1.5, 2.1)
+        img = torch.rand(1, 3, 5, 5, device=device, dtype=dtype)
+        aug = RandomGaussianBlur(kernel_size, sigma, "replicate")
+        expected = aug(img)
+        aug = aug.compile(fullgraph=True)
+        actual = aug(img)
+        assert actual.shape == img.shape
+        self.assert_close(expected, actual)
```
Comment (johnnv1) [inline tests/augmentation/test_augmentation.py:3864]: I wondered, shouldn't this fail due to the randomness of the augmentations?

Comment (edgarriba) [reply tests/augmentation/test_augmentation.py:3864]: i'll check

**Pull request merged**